### ISSUE TITLE
Enable Bold Text in Windows Terminal

### ISSUE THREAD

**Issue by bitcrazed** [microsoft/terminal#109]
_From @agocke on April 27, 2017 22:5_

Repro: `echo -e "text \033[1mbold\033[0m text"`

This prints the following for me:

![image](https://cloud.githubusercontent.com/assets/515774/25506272/aa183300-2b5a-11e7-896d-4605911466f2.png)

`bold` should actually be bold, not just another color in the color palette. It also shouldn't change the foreground color.

Microsoft Windows [Version 10.0.15063]

_Copied from original issue: Microsoft/WSL#2027_

**Comment by bitcrazed** [comment]
_From @sunjoong on April 27, 2017 23:17_

@agocke - lt looks not bold but __bright__. Effect of code 1 is "Bold or increased intensity -- [ANSI escape code](https://en.wikipedia.org/wiki/ANSI_escape_code)".

**Comment by bitcrazed** [comment]
_From @zadjii-msft on April 27, 2017 23:24_

@agocke that's exactly right. We don't support proper bold at the moment, but the intensity works just fine for now. Like with all feature requests, it's already on the backlog :)

**Comment by bitcrazed** [comment]
_From @agocke on April 27, 2017 23:29_

@zadjii-msft How do you decide what "intensity" means in this context? It seems to pick a different color from the palette, i.e. I can actually change the color that is printed by bold and it can be completely different from the normal color.

This is important because if I change the color scheme for my 8 foreground colors, what happens to my bold colors? It seems like sometimes nothing may happen, which means the bold colors show up as unreadable in my new color scheme.

**Comment by bitcrazed** [comment]
_From @fpqc on April 28, 2017 3:6_

@agocke This is pretty standard for terminal emulators.  The 'bright' colors are the second row of 8 colors in programs that have a sane color picking interface =p.

**Comment by bitcrazed** [comment]
_From @agocke on April 28, 2017 3:26_

Yeah, I think I screwed up the transcription from my .Xresources. Still, bold would be useful.

**Comment by bitcrazed** [comment]
_From @dgilmour77 on November 14, 2017 15:45_

This creates a bit of a trap for unwary users.  I like black text on dark grey background, and when I set it, it looked great.  Until I brought up a man page and it looked garbled and broken because I was unlucky that the bold color automatically chosen was the same as my background color - dark grey.  At a minimum, the logic for choosing a bold color should be based not just on the foreground color but the foreground/background combination.  Yeah, good luck with that ;-)

**Comment by j4james** [comment]
If you do decide to implement this, is there any chance it could be made an option? Using the SGR bold (increased intensity) parameter to generate 16 colors has been a de facto standard (at least in the DOS world) for what must be decades now. And I think the default Linux console might work that way too actually. It would be sad to lose that functionality if you change it to bold instead.

I realise there are there are now other ways of getting bright colors, but that's obviously going to be a breaking change for old code, not to mention all that glorious ANSI artwork from the BBS era.

**Comment by bitcrazed** [comment]
@agocke: "Still, bold would be useful"

Bold is a little tricky because it actually changes the physical size & shape of letters and, depending on what fonts you use, sometimes ends up looking pretty awful.

Alas, the Console's current text renderer uses GDI which is > 30 years old and lacks much of the sophistication of modern text rendering technologies like DirectWrite.

Console's GDI renderer can't currently handle bold text: 
![image](https://user-images.githubusercontent.com/961950/37857709-b1d3404e-2eba-11e8-843c-60cd6bbc4f8d.png)

But VSCode's integrated terminal takes advantage of Chrome which takes advantage of DirectWrite:
![image](https://user-images.githubusercontent.com/961950/37857640-f9d46ec8-2eb9-11e8-8d53-16e17614eaa2.png)

We are planning on replacing our GDI renderer with a DirectWrite renderer in the future which will allows us to (finally) render bold text, emoji, and all manner of other beautiful text effects, but until then, the best we can do is display "bold" text with a more intense color.

HTH.

**Comment by thchr** [comment]
I don't suppose there's any chance that the specific time of the 'future' could be teased here: might we see an insider build with DirectWrite in 2018 or 2019? 

From an aesthetic point of view, it would be great if Windows could catch up with most standard Unix consoles in being able to print formatted text (improving text-display in WSL and some REPLs (e.g. in Python & Julia) on Windows)

**Comment by zadjii-msft** [comment]
Boy it would be great if I could give out exact timelines, but there are a lot of factors that affect how long a feature takes to get out the door.

I'd rather say nothing at all rather than give a timeline and let it slip.

**Comment by yveslange** [comment]
Is this still planned ? Any news ? <3

**Comment by zadjii-msft** [comment]
Yep, still on the backlog. We'll make sure to post here when there's something to share :)

**Comment by leifmetcalf** [comment]
On a light background, the bold text is basically invisible. With the "One Half Light" theme it's just barely lighter than the background, and with the "Solarized Light" theme it's the same shade as the background.

**Comment by antoineco** [comment]
@leifmetcalf probably due to #293

**Comment by qqkookie** [comment]
I have same problem with Bold attribute of character.  Using WT preview build and Windows 10 + WSL 2 Ubuntu nano editor. I set nano to use bold face instead of reverse video character attribute.  then the text become invisible on light color theme screen. On dark color theme it is displayed in normal face. I tested it with various fixed width fonts like Consolas, Lucida Console, D2Coding, same problem.

Interestingly, with dark color theme ("Scheme" like "Campbell", "One Half Dark"), the bold characters are displayed anyway in normal face (not bold), but with  Light color Scheme like "Solarised Light" or "One Half Light", the bold characters become totally invisible. 

I suspect some problem related to this  #571 (Some colors are swapped in Terminal), problem in  color mapping.

So I suggest that the bold characters should be rendered in normal face with light color scheme,  same as it does with dark color scheme, as interim stopgap solution until it is fixed completely.

**Comment by leifmetcalf** [comment]
I used to edit the 'One Half Light' colourscheme so that the 'bright white' colour was black, which fixed it for me. Dunno how to do that now that the default colourschemes aren't in the config.

**Comment by qqkookie** [comment]
That may be reason that reverse video character attribute does not work in Light color scheme. 

In dark color scheme, the reverse video does work correctly ( character => background color(black), character background +> foreground color (white) )  But in light color scheme, doing same processing for reverse video does not work, because  the fore/back assignment combination is same as normal fore/background combination of light color scheme (i.e character => black, background => white) .  So revere video text part does not stand out from normal text.  Correct processing should do the opposite  (i.e character =>background (white in Light scheme) , character background => (black in Light scheme)  in Light color scheme.  There seems that the reverse video part of the code is using wrong colors, using color-0/color-16 directly instead of foreground/background color indirectly. Reverse video should use color specified as background color for charter color  and foreground color for filling character background.

**Comment by jdebp** [comment]
[`Attributes.ps1`](https://github.com/jdebp/terminal-tests/blob/master/PowerShell/Attributes.ps1) shows the level of attribute support in Windows Terminal.  Currently, several things are missing, or implemented in the way that M. Turner noted earlier.

**Comment by Qix-** [comment]
>  @agocke - lt looks not bold but bright. Effect of code 1 is "Bold or increased intensity -- ANSI escape code".

I believe the wikipedia page states that _because_ microsoft did it wrong for so many years. Most implementations I've ever come across change the glyph weight without touching the foreground color.

Perhaps Microsoft can retire the old cmd.exe behavior. This would be a good chance to make things right.

**Comment by heftig** [comment]
> I believe the wikipedia page states that _because_ microsoft did it wrong for so many years. Most implementations I've ever come across change the glyph weight without touching the foreground color.

I took a sample of the terminals available to me, in their default configuration:

PuTTY - only bright
Linux fbcon - only bright
XTerm - both bright and bold
urxvt - both bright and bold
Konsole - both bright and bold
Terminology - both bright and bold
VTE (GNOME Terminal) - only bold (past versions used to be both bright and bold)
Alacritty - only bold
OSX Terminal - only bold
iTerm2 - only bold

**Comment by zadjii-msft** [comment]
oh man, I guess I'll summon @egmontkob and @j4james into this thread as well, to maybe summarize #5682/#5384/#2661

**Comment by j4james** [comment]
Yeah, I don't want to get into another long argument about this. I just want to say two things:
1. A lot of the statements that have been made on this subject are simply false.
2. At some point after #2661, assuming this issue is still open, I would like to submit a PR which adds bold font weight support for the ITU indexed and RGB colors, which I think is a reasonable compromise (and I believe would match Mintty's default behaviour). That's as far as I'm willing to go.

**Comment by agocke** [comment]
As the original filer of this issue I'll note that I did screw up porting my Xresources and the behavior after I fixed that was fine.

**Comment by egmontkob** [comment]
The recent trend is to fully separate the font weight from the color. Foreground color is set by SGR 30-38 and 90-97 (39 or 0 turns it off); whereas bold lives on as the previously ambiguous bold-or-bright SGR 1 (22 or 0 turns it off). [It's a common misbelief that 21 turns it off; no it doesn't, it's 22.]

This makes a lot of sense ever since 256-color and later truecolor support was introduced in many terminals, and is also the mode required for the popular Solarized theme. I believe this mode is the only reasonable way to get rid of legacy's no longer relevant limitations and to move forward. This behavior comes with backwards compatibility issues, though, even in the Linux world, and probably much more prominently in the Windows world.

For VTE, the discussions are mainly at bugs [762247](https://bugzilla.gnome.org/show_bug.cgi?id=762247) (this one is more about the current technical side) and [791596](https://bugzilla.gnome.org/show_bug.cgi?id=791596) (this one is more about the legacy and the philosophical side).

What in my opinion every terminal should do:

- Internally, track for each character cell the SGR 1 state, and independently from this, the SGR 3x/9x state. (And while at it, SGR 3 for italic.)

- Implement painting bold typeface. (And while at it, SGR 3 for italic.) [It's indeed a bit unfortunate that there's no additional room for bold chars, it's up to each terminal to fine tune the look, e.g. pick the bold characters' width for the cell, or the average of the regular and bold, or whatever that looks the best. In VTE we pick the regular characters' width, thus bold text is a bit squeezed.]

- Implement a couple of possible mappings from the internal logical SGR 1 and 3x/9x states to the physical boldness and color display mode; including one or more traditional behaviors where SGR 1 brightens the legacy first 8 colors, maybe boldens everything, maybe doesn't use bold typeface anywhere at all, etc., and also a trivial direct mode where SGR 1 specifies the font weight and SGR 3x/9x specifies the color without overhearing.

**Comment by Qix-** [comment]
@j4james 

> A lot of the statements that have been made on this subject are simply false.

I mean, meh. There are no relevant standards to refer to for today's emulators. Linguistically, "bold" does not mean changing the color. That's always been a Microsoft behavior, in my head at least.

**Comment by rbeesley** [comment]
@leifmetcalf

> I used to edit the 'One Half Light' colourscheme so that the 'bright white' colour was black, which fixed it for me. Dunno how to do that now that the default colourschemes aren't in the config.

You can use the shortcut to open the default settings profile, and then introduce a new scheme to your settings which override the defaults, copying the values from the default. The way the settings are applied, this will allow you to make changes. Similarly, if you don't want to directly override the default scheme, you can create a new scheme using those values and then apply them to the profiles you want to modify.

**Comment by zachmatson** [comment]
I find myself just going into VS Code for the terminal much of the time because of this... It renders bold text perfectly

**Comment by tigerinus** [comment]
> > I believe the wikipedia page states that _because_ microsoft did it wrong for so many years. Most implementations I've ever come across change the glyph weight without touching the foreground color.
> 
> I took a sample of the terminals available to me, in their default configuration:
> 
> PuTTY - only bright
> Linux fbcon - only bright
> XTerm - both bright and bold
> urxvt - both bright and bold
> Konsole - both bright and bold
> Terminology - both bright and bold
> VTE (GNOME Terminal) - only bold (past versions used to be both bright and bold)
> Alacritty - only bold
> OSX Terminal - only bold
> iTerm2 - only bold

VSCode Integrated Terminal - only bold

**Comment by rbeesley** [comment]
@tigerinus, while I suppose that could be why Wikipedia says that, that seems unlikely to be the real reason. `ANSI.SYS` was necessary to use any of the SGR attributes like intensity. So I don't know why that would have been the case for other POSIX terminals.

The standard just has SGR 1 as increased intensity or bold, so it's pretty open. Interestingly 90-97 for bright foreground and 100-107 for bright background isn't actually part of ECMA-084 that I could see.

One of the benefits of using SGR 1 for bold instead of increasing the intensity is that it would provide a lot more control over the output. As it is right now SGR 1 and 90-97 effectively overlap with the same result. As long as the non-standard attributes provide access to the bright foreground and background, SGR 1 is less useful on its own.

**Comment by gordonmessmer** [comment]
I wanted to expand @heftig 's results, with a few corrections.  (I can provide screenshots for most of these if they would be helpful)

I compared the output of colored text, bold colored text, bright colored text, and bold bright colored text with this command:

for x in 3 9 ; do for y in $(seq 0 7) ; do echo -e "\033[0;${x}${y}mblah \033[1;${x}${y}mblah" ; done ; done

Terminal | Bold | Bold and Bright
-- | -- | --
alacritty | ok | ok
gnome-terminal | ok | ok
kitty | no [1] | no [1]
konsole | ok | no [2]
mate-terminal | ok | no [2]
rxvt | ok | no [2]
terminator | ok | ok
xfce4-terminal | ok | no [2]
xterm | ok | no [2]
Linux console | no [3] | no [2]
OS X Term | ok | ok
PuTTY | no [3] | ok
Windows Term | no [3] | no [2]

1: bold has no effect
2: bold and bright is the same as bold
3: bold is treated as bright

All of the terminals tested supported a standard and a bright color palette.

Of the terminals tested, other than Windows Terminal, only the Linux FB console and PuTTY (and its derivative mintty) will used the bright color palette for bold colored text.

However, Windows Terminal has one unique flaw.  Windows Terminal is the only terminal tested that (as best as I understand its behavior) uses "bright white" text when the bold attribute is supplied alone, regardless of the color of the background.  This makes all of the light-background themes supplied with Windows Terminal unusable, or very bad.  When using common tools like "top", many fields are difficult or impossible to read.

Lacking support for bold text, it would be much better in the short term if Windows Terminal simply ignored bold, the way that kitty does.

Later, when bold text support is added, Windows Terminal should adopt the common behavior of using the standard color palette for bold colored text, and bright color palette only when it is specified.

**Comment by DHowett** [comment]
@gordonmessmer i recall seeing a “bright default” color entry in gnome-terminal some months ago, for use in the `1;39` case. Have you seen anything like that in the terminal emulators you’ve profiled?

**Comment by gordonmessmer** [comment]
"i recall seeing a “bright default” color entry in gnome-terminal some months ago, for use in the 1;39 case. Have you seen anything like that in the terminal emulators you’ve profiled?"

Do you mean the "bold color" preference?  I didn't look for that setting in the other terminals.  That seems like a preference that only makes sense for terminals that don't support bold weight fonts.  In #5682, @egmontkob suggested that the preference doesn't make sense any more and was likely to be deprecated.

I can go through them again if it would help.

"The specific behavior you're seeing ... is a legacy compatibility behavior"

I'm sure.  My purpose was to demonstrate that WT is nearly alone in implementing that behavior.  gnome-terminal supports it as an optional behavior, and other terminals probably do too.  I didn't look.

I guess I'd have to ask whether WT is intended to be a legacy support tool, or in the words used in the README, "a new, modern, feature-rich, productive terminal application".  The current behavior is at odds with that description, and if the current behavior is a bug, then that's fine.  Reasonable people don't expect anything to be perfect, especially in early releases.  However, some of the bug reports that I've read suggest that at least one developer is unwilling to consider supporting the standard behavior, and that is disappointing.

**Comment by DHowett** [comment]
As I’m sure you’re aware, it’s never quite so crisply-defined into buckets like “legacy support software” or “shiny new thing.”

Without color elision, every application written targeting the console API looks terrible (https://github.com/PowerShell/PSReadLine/issues/830#issuecomment-650508857 or some variation thereof; CMD gets more black bars, powershell gets fewer, FAR demolishes your buffer). Like as not, those applications are our “legacy”. I don’t mean that as a perjorative—there are applications being written today that don’t or can’t or won’t use VT, and they are an important part of our ecosystem.

Of course, that doesn’t mean we aren’t improving what we can. #6506 and #6698 are a huge step in the right direction; with those, and thanks entirely to @j4james, we can finally tell the difference between 39 and 37 and 97 and the Win32 API `SetConsoleTextAttribute(0x07)`. Once those land, we’ll be in a better place to make plans around supporting typographical bold versus brightening.

Which one we’ll end up doing and how configurable it will be should be the topic of a specification and not a GitHub issue comment written up on my phone.

Regardless, this request wouldn’t still be open if we weren’t going to do _something_ about it.

**Comment by j4james** [comment]
> @gordonmessmer i recall seeing a “bright default” color entry in gnome-terminal some months ago, for use in the `1;39` case. Have you seen anything like that in the terminal emulators you’ve profiled?

I know XTerm, VTE, Mintty, Tera Term, iTerm2, PuTTY, and VTStar all have an option like that.

**Comment by gordonmessmer** [comment]
To be clear, I regard the inability to render text with bold font weight as a superficial issue.  I also regard the conversion of bold+color to bright color as a superficial issue.

However, converting bold alone to bright white unconditionally seems like a bug if users can configure a light color background.  I'm not sure this is the right bug report in which to continue that discussion.  #5682 might be, but I dislike the practice of framing a bug in terms of one possible solution.

I see your screenshots in #6698 and I can't tell if that change is an improvement to this situation, because it looks like you're only testing the change in terminals with a dark background.  If you're not testing any of the light themes, I think you're likely to continue to blame a minor flaw in Solarized Light for a significant flaw in Terminal.

**Comment by j4james** [comment]
@gordonmessmer Below are some screenshots showing the effects of the #6506/#6698 PRs on the bold attribute in a couple of light color schemes. Comparing before and after (first and second screenshot from each scheme) you can see the bold attribute is no longer mapping to white. That was simply a conpty narrowing bug. And while the default bold color interpretation may still have room for improvement, that's best discussed in #5682.

![image](https://user-images.githubusercontent.com/4181424/85955775-fe301700-b978-11ea-9fe9-527b3fc0b04d.png)

**Comment by edyu** [comment]
For people who use terminal versions of vim and nvim, it would be super useful to have this "feature" implemented or this bug fixed along with the problem with italics.

**Comment by swinder0161** [comment]
![image](https://user-images.githubusercontent.com/397543/93730072-ac4c0580-fbe4-11ea-93b4-db8eeb20ae56.png)
Fluent Terminal on windows can show Bold + Bright

![image](https://user-images.githubusercontent.com/397543/93730122-e6b5a280-fbe4-11ea-915b-ddf20ab167d4.png)
Windows terminal seems to do nothing

**Comment by DHowett** [comment]
@swinder0161 yes, that is the basic premise of this _open issue_ titled “enable bold text in Windows Terminal”. If you want to register your support for it, consider clicking on the 👍 button below the top comment.

**Comment by bsidhipong** [comment]
+1 for enabling this feature, please?  It makes reading manpages and other day-to-day tasks on the more Linux systems enjoyable.
![image](https://user-images.githubusercontent.com/1375277/95658590-2f27f800-0ae1-11eb-881d-470e7c97b7bd.png)

**Comment by zadjii-msft** [comment]
@bsidhipong If you'd like to "+1" this feature, the best way to do that is by hitting the 👍 button on this issue

![image](https://user-images.githubusercontent.com/18356694/96140789-ef3c8880-0ec5-11eb-82e0-30ce4977b9a7.png)

That way, you avoid unnecessarily pinging everyone following this thread. Thanks!

**Comment by DanielAtKrypton** [comment]
I was able to achieve the functionality I wanted with Vs Code, bold is working perfectly fine and rendered as `DOM`. It would be great to have the same possibility with windows terminal.

[More information](https://github.com/tonsky/FiraCode/issues/1161)

**Comment by jm33-m0** [comment]
I can't believe this is still open after almost 3 years...
When using `lazygit` in Windows Terminal, bold has absolutely no effect, causing the cursor location undetectable.
I have to add that in default console of powershell, `lazygit` works quite well, with bright color as bold

**Comment by MrInfusion** [comment]
> I can't believe this is still open after almost 3 years...
> When using `lazygit` in Windows Terminal, bold has absolutely no effect, causing the cursor location undetectable.
> I have to add that in default console of powershell, `lazygit` works quite well, with bright color as bold

When do Windows Terminal Devs implement that? It does take so long and still nothing. Cant they just switch to DirectWrite? Is it that hard?

**Comment by DanielAtKrypton** [comment]
> 
> 
> > I can't believe this is still open after almost 3 years...
> > When using `lazygit` in Windows Terminal, bold has absolutely no effect, causing the cursor location undetectable.
> > I have to add that in default console of powershell, `lazygit` works quite well, with bright color as bold
> 
> When do Windows Terminal Devs implement that? It does take so long and still nothing. Cant they just switch to DirectWrite? Is it that hard?

There might be not enough community appeal to develop that at the moment. The more people vote [here](https://github.com/microsoft/terminal/issues/109#issue-297948117), the more likely devs will focus on this feature rather than on others. The very same technology VS Code uses could be used here too, I believe. Maybe someone could dehydrate VS Code to make a standalone terminal too.

**Comment by mdtauk** [comment]
> > > I can't believe this is still open after almost 3 years...
> > > When using `lazygit` in Windows Terminal, bold has absolutely no effect, causing the cursor location undetectable.
> > > I have to add that in default console of powershell, `lazygit` works quite well, with bright color as bold
> > 
> > 
> > When do Windows Terminal Devs implement that? It does take so long and still nothing. Cant they just switch to DirectWrite? Is it that hard?
> 
> There might be not enough community appeal to develop that at the moment. The more people vote [here](https://github.com/microsoft/terminal/issues/109#issue-297948117), the more likely devs will focus on this feature rather than on others. The very same technology VS Code uses could be used here too, I believe. Maybe someone could dehydrate VS Code to make a standalone terminal too.

Terminal does use Direct Write AFAIK - and support for Bold text is being worked on.

There are some complications when a user chooses a bold weight font in their settings, which need working out.

And not every feature in Windows Terminal, can be ported back to the version of console that is backwards compatible

**Comment by j4james** [comment]
I actually hacked together a POC implementation of this in conhost a while back - it doesn't require a lot of code. I didn't look at the DX engine, but I can't imagine it'd be that much more effort.

And you're right about the complications when you start with a bold font. I just did a quick test now with various weights of Cascadia - the _SemiBold_ version showed no difference when having a heavier weight applied to it as far as I could tell. It's possible that's just a limitation of that font though, or maybe the GDI renderer.

You can see some screen shots below demonstrating the behaviour with different weights: default, _ExtraLight_, and _Semi-Bold_.

![image](https://user-images.githubusercontent.com/4181424/102000343-86636800-3cde-11eb-8fe5-388243523fa5.png)

**Comment by whytong** [comment]
> I can't believe this is still open after almost 3 years...
> When using `lazygit` in Windows Terminal, bold has absolutely no effect, causing the cursor location undetectable.
> I have to add that in default console of powershell, `lazygit` works quite well, with bright color as bold

mintty/ComEmu have bold font implemented like 5-10 years ago?

**Comment by TBBle** [comment]
> Terminal does use Direct Write AFAIK - and support for Bold text is being worked on.

Am I correct in understanding that a needed bold text support feature in DirectWrite blocks Windows Terminal from supporting bolded text for inline `\e[1m`?

There haven't been any [updates to DirectWrite since Windows 10 1903](https://docs.microsoft.com/en-au/windows/win32/directwrite/what-s-new-in-directwrite-for-windows-8-consumer-preview). Is bold text support still being worked on for DirectWrite?

Or is future DirectWrite feature work now focussed on [DWriteCore](https://docs.microsoft.com/en-au/windows/win32/directwrite/dwritecore-overview) and so Windows Terminal will have to migrate to it to gain new text rendering features, once it is up-to-spec?

**Comment by DHowett** [comment]
Nope, this is totally blocked on us. DWrite is an excellent piece of software and I hope that it never changes (because changes inevitably break things for us :smile:).

@skyline75489 is slowly marching towards bold text support -- we just haven't been reviewing his PRs fast enough.

**Comment by skyline75489** [comment]
Ideally this is a simple feature, but right now the underlying font handling code needs a bit of refactoring to better support bold & bold italic. I've filed #9096 and #9201 to support the work. Hopefully bold will be supported before/at version 2.0. Personally I love to see it supported, because it's the last eye-candy thing that's missing in WT, comparing to all the well-known Linux terminals.

**Comment by swinder0161** [comment]
Ditch the WT. And use gnome terminal for wslgui

**Comment by j4james** [comment]
> This is basic vt100/ansi.sys compliance

@riedel Not that it matters, but the DEC VT100 did not support bold font-weights, nor did any other DEC terminal that I'm aware of (that setup screen in your screenshot is not from a real DEC VT132 terminal). And ANSI.SYS is a DOS device driver which didn't support bold font-weights either. So if you're arguing in favor of strict VT100/ANSI.SYS compliance, you're really arguing *against* supporting bold fonts. 😉

This is a popular feature request, and it *is* going to get implemented. Just be patient. You don't need to invent fake arguments to prioritize it.

**Comment by riedel** [comment]
@j4james: thanks, you are totally right. The escape I think was introduced by SCO Unix or so. I admit that I was wrong here: still this no eyecandy. I guess I was remembering bright fonts on my vt520s . I think my mistake in memory was that bright was actually always looking bold on the old terminals. 
![grafik](https://user-images.githubusercontent.com/566485/117533681-5ea6d080-afee-11eb-9c01-170f7fb3a244.png)

Still it is not eye-candy: Bold is  IMHO necessary to emulate bright on modern monitors for white fonts. 

It is not a fake argument, sorry. Again my appologies for my wrong memory.

**Comment by swinder0161** [comment]
gnome-terminal using wsl gui. If you love gnome terminal, this works flawlessly thanks to windows wslgui support.
see it running Powershell, cmd and ubuntu shell

![image](https://user-images.githubusercontent.com/397543/117565817-f0870a00-b0d0-11eb-872f-d2c1ea636c2e.png)

**Comment by skyline75489** [comment]
Ladies and gentlemen, we are one step closer to the goal now. I've filed #10498 for this. 

I'm here to encourage anyone who's interested in this feature to take a look at that PR. Even though I love the feature, I'm no expert on font or VT history staffs. So I need the help from you guys.

Thanks again for all the attention & discussion!

**Comment by TBBle** [comment]
Looking good. Using https://github.com/jdebp/terminal-tests/blob/master/PowerShell/Attributes.ps1:

![image](https://user-images.githubusercontent.com/138397/125676264-7e5626ef-6202-48e2-939a-8ce56dfaeb90.png)

Since it's a picture, I'll note that both rapid and normal blink are the same speed (probably as-intended) but 'no rapid blink' is blinking as well, suggesting either a bug in the test script, or a failed state-clear somewhere in Terminal. But neither is a problem for this bug.

**Comment by riedel** [comment]
Bold works for me now on cygwin/wsl2 bash after setting ```intenseTextStyle": "bold"```  (just as a comment for anyone who stumbles on this and wants to try)

**Comment by Konfekt** [comment]
That is, add the line of `"intenseTextStyle": "bold",` to every item in the array `profiles.list` in the file `settings.json`.

**Comment by TBBle** [comment]
@Konfekt It doesn't work if you add it to the profile defaults? That sounds like a bug that should be raised as a new issue, rather than as a comment on a closed MR. Also, this MR actually _predates_ it being selectable, so it's not even a relevant MR for your concern.

**Comment by Konfekt** [comment]
It is me who did not understand the GUI setting, which says "Text formatting" with options "normal, bold, intense, bold and intense". I thought for example that the option "bold" changes the default text format to bold, rather than that it enables rendering ANSI escape codes for bold as bold.

This said, I wonder how to enable italic fonts under Vim in Windows? These work perfectly in Vim in WSL, but not in Vim in Powershell or cmd.

**Comment by zadjii-msft** [comment]
> This said, I wonder how to enable italic fonts under Vim in Windows? These work perfectly in Vim in WSL, but not in Vim in Powershell or cmd.

This is probably a limitation of `vim.exe`. They're likely using the Win32 API to format the text (which doesn't support italics) vs the VT sequences that WSL vim would use (which does support italics)

**Comment by Konfekt** [comment]
Thank you, this is now filed as a feature request in the Vim issue tracker.

**Comment by kbridge** [comment]
> That is, add the line of `"intenseTextStyle": "bold",` to every item in the array `profiles.list` in the file `settings.json`.

Or add this setting to `profiles.defaults` object.

**Comment by avih** [comment]
Is there a way to make the default bold color render as bright, like the first message on this issue:
![aa183300-2b5a-11e7-896d-4605911466f2](https://github.com/microsoft/terminal/assets/2164962/ea104a7b-1d52-4e90-b0d8-e8750d82a15a)

I'd be fine if it's emboldened too, but I do want it brightened, because bold alone doesn't stand out enough (for my taste) , but brightened color does stand out nicely.

I do have `"intenseTextStyle": "all",`, but this seem to apply to the colors 0-7, but not to the default foreground color.

I.e. `printf "\033[37m white \033[1m bold and bright\033[m\n"` is white (7) and then bold-white which is rendered as both bold and brightened (that's great).

But this `printf "default \033[1m bold only :( \033[m\n"` only does bold, without bright.

One could argue that the most commonly SGR-1 color is the default foreground color, so it could be useful if the `intenseTextStyle` option could apply to the default color too.

It could probably either be automatically brightened, or as an additional color at the theme (near the default color).

Alternatively, can I set white (SGR 37) as the default color, so that SGR-1 on the default color ends up respecting `intenseTextStyle` ?

**Comment by TBBle** [comment]
I'd suggest opening a new feature request for this, as this ticket is closed.

I suspect what you want is "hard", because the "bold is bright" logic is to use the "bright" version of the colour, i.e. the second 8-colour row of the 16-colour palette, the same as adding 60 to the ANSI colour code.

However, that logic doesn't apply for the default colour (which is not an entry in the palette, but a separate slot) so we'd need a way to define "bright" for that colour specifically. It might make sense to be able to apply such generally, hence being able to enbrighten more than just the base 8 colours when using bold-is-bright. In cases like Solarized, where some implementations rely on bold-is-bright to access the full 16-colour palette, or just for users who expect the pre-defined 8 "bright" colours in the palette to apply to 'bold is bright' too, this logic does not apply, so this is a new option, it can't be a global change.

Conversely, detecting when the default colour is one of the non-bright 8 palette colours, and brightening it along with its palette-mate, would possibly resolve your issue. That's also a new feature...

So yeah, "hard", not an obvious or trivial fix.

**Comment by DHowett** [comment]
Funny enough, this is on our radar!

Right now, _if your foreground matches one of the colors from 0-7_ we will automatically "brighten" it using the equivalent indexed color from 8-15!

In the future, we're going to offer a specific "brightForeground" color specifier so that color schemes can indicate exactly how to brighten the foreground. That is tracked in #5682.

**Comment by avih** [comment]
> I'd suggest opening a new feature request for this, as this ticket is closed.

Well, it was implemented originally, which is why this issue was opened in the first place, so I was merely asking whether it's still possible to do that ;)



> Funny enough, this is on our radar!

Glad to hear!

> Right now, _if your foreground matches one of the colors from 0-7_ we will automatically "brighten" it using the equivalent indexed color from 8-15!

That's actually quite good enough for me. Much appreciate the info!
Also, confirmed working! whooo :)

> In the future, we're going to offer a specific "brightForeground" color

Right, that's what I thought would be the best general solution too.

> That is tracked in #5682.

I'll follow that issue.

Thanks again!

**Comment by nad2000** [comment]
For the record keeping adding this here, I had the same issue with (Version: **1.22**.11141.0 :).  The solution working with WSL (tmux, vim, etc) was to change "Appearance -> Text Formatting -> Intense text style" to "Bold font with bright colours":

![Image](https://github.com/user-attachments/assets/115b1c61-9dfc-467a-a649-9b82fc4afdcf)
